File: web/src/app/core/models/venta.ts
import { Cliente } from './cliente';
import { Empleado } from './empleado';

export class Venta {
  constructor(
    public idVenta: number = null,
    public cliente: Cliente = null,
    public empleado: Empleado = null,
    public numeroSerie: string = null,
    public fechaVentas: Date = null,
    public monto: number = null,
    public estado: string = null
  ) {}
}

File: src/main/kotlin/com/bykenyodarz/sistemasventasweb/models/Venta.kt
package com.bykenyodarz.sistemasventasweb.models

import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "ventas")
class Venta {

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "IdVentas")
    var idVenta: Int? = null

    @OneToOne
    @JoinColumn(name = "IdCliente")
    var cliente: Cliente? = null

    @OneToOne
    @JoinColumn(name = "IdEmpleado")
    var empleado: Empleado? = null

    @Column(name = "NumeroSerie")
    var numeroSerie: String? = null

    @Column(name = "FechaVentas")
    var fechaVentas: LocalDate? = null

    @Column(name = "Monto")
    var monto: Double? = null

    @Column(name = "Estado")
    var estado: String? = null

    @PrePersist
    fun prePersist() {
        this.fechaVentas = LocalDate.now()
    }

}
